packages/types/src/lib/interfaces.ts

```diff
@@ -1094,15 +1094,44 @@ export declare type AuthenticatorAttachment = 'cross-platform' | 'platform';
 /**
  * ========== PKP ==========
  */
+export interface LitClientSessionManager {
+  getSessionKey: () => SessionKeyPair;
+  isSessionKeyPair(obj: any): boolean;
+  getExpiration: () => string;
+  getWalletSig: (getWalletSigProps: GetWalletSigProps) => Promise<AuthSig>;
+  // #authCallbackAndUpdateStorageItem: (params: {
+  //   authCallbackParams: AuthCallbackParams;
+  //   authCallback?: AuthCallback;
+  // }) => Promise<AuthSig>;
+  checkNeedToResignSessionKey: (params: {
+    authSig: AuthSig;
+    sessionKeyUri: any;
+    resourceAbilityRequests: Array<LitResourceAbilityRequest>;
+  }) => Promise<boolean>;
+  getSessionSigs: (params: GetSessionSigsProps) => Promise<SessionSigsMap>;
+  signSessionKey: (
+    params: SignSessionKeyProp
+  ) => Promise<SignSessionKeyResponse>;
+}
+
+export interface AuthenticationProps {
```

Comment:
this looks good but i think we also need `authNeededCallback` for if they wish to override the default implementation with out and explicit call to `getSessionSigs`